dmoj/cptbox/isolate.py

```diff
@@ -247,26 +248,52 @@ def _file_access_check(self, rel_file, debugger, is_open, flag_reg=1, dirfd=AT_F
         # without any downside, and if it was *not* NULL and we failed to read
         # it, then we should *definitely* stop the call here.
         if rel_file is None:
-            return '(nil)', False
+            return '(nil)', ACCESS_EFAULT
+
+        is_write = is_open and self.is_write_flags(getattr(debugger, 'uarg%d' % flag_reg))
+        fs_jail = self.write_fs_jail if is_write else self.read_fs_jail
 
         try:
             file = self.get_full_path(debugger, rel_file, dirfd)
         except UnicodeDecodeError:
             log.exception('Unicode decoding error while opening relative to %d: %r', dirfd, rel_file)
-            return '(undecodable)', False
+            return '(undecodable)', ACCESS_EINVAL
 
-        is_write = is_open and self.is_write_flags(getattr(debugger, 'uarg%d' % flag_reg))
-        fs_jail = self.write_fs_jail if is_write else self.read_fs_jail
-        if fs_jail.match(file) is None:
-            return file, False
-        return file, True
+        actual = normalized = '/' + os.path.normpath(file).lstrip('/')
```

Comment:
Since this is a hack, let's at least leave a comment for why we're doing all this.

Reply:
Also, could we name the variables something a little more clear? We have `actual` `real` `normalized`, and the first two are synonyms of each other in English... I will confess to not being able to quickly grok this logic because of this.

Reply:
Fixed.